src/coreclr/vm/assembly.cpp

```diff
@@ -1499,8 +1564,42 @@ MethodDesc* Assembly::GetEntryPoint()
     }
     else
     {
+#if defined(TARGET_BROWSER)
+        // we only support IL entrypoints in browser
+        COMPlusThrowHR(COR_E_BADIMAGEFORMAT);
+#else // !TARGET_BROWSER
```

Comment:
```suggestion
```
This ifdef should not be needed. The path handles global function IL entrypoint - the comment is misleading. 

You cannot generate these using C#, so this path is probably not going to be reached in practice. We may have some tests that hit it.